### ISSUE TITLE
Update Semtech LoRaMAC to latest version

### ISSUE THREAD

**Issue by jcwren** [RIOT-OS/RIOT#16105]
#### Description

RIOT supports LoRaWAN class A and class C, but not class B. Semtech / Stackforce doesn't like to put version numbers in their source code for some reason, but I know it's from before class B had proper support, as there is no reference to the beacon channels nor anything else related to full class B support. 

I'm too new to RIOT to integrate the latest release of the LoRaMAC code, otherwise I'd take a stab at it.

I was able to get a simple class A and class C app that both transmits and receives running on the ST B-L072Z-LRWAN1 board fairly quickly. 

### Useful links

https://github.com/Lora-net/LoRaMac-node

**Comment by akshaim** [comment]
Yes, you are right current Semtech package used in RIOT is version 4.4.1 which is not fully in compliance with the 1.0.3 spec. Migrating to 4.4.2 or newer is required. Added this to-do list :)
